x-pack/filebeat/module/suricata/eve/ingest/pipeline.json

```diff
@@ -93,7 +100,39 @@
       , "value": "{{suricata.eve.event_type}}"
       }
     }
-
+  , {"convert":
```

Comment:
Do we need type conversion? Otherwise we could just use the rename processor: https://www.elastic.co/guide/en/elasticsearch/reference/current/rename-processor.html

Reply:
Good point. I wonder why it was done this way. Did I do it this way initially? (If so, I'm sorry, this was my very first module LOL)

Reply:
Never mind, it just occurred to me that this was the simplest way to copy the data out (instead of renaming).

We had decided that since Suricata had a JSON format which is already very familiar to people, we would start for now by leaving the full untouched event in place, and only copy out data, instead of renaming the fields.

This is different from most modules, but most modules have field names determined by the grok patterns (us), not determined by the tool creator (and familiar to the tool's community).

(My first coffee just kicked in, haha)

Reply:
Back in the day `convert` was the only way we found to copy a field and failing silently if the source field doesn't exist.

That, or an ugly gigantic piece of painless.